lisa/sut_orchestrator/azure/features.py

```diff
@@ -2464,3 +2465,56 @@ def create_setting(
             return schema.FeatureSettings.create(cls.name())
 
         return None
+
+
+class PasswordExtension(AzureFeatureMixin, features.PasswordExtension):
+    @classmethod
+    def create_setting(
+        cls, *args: Any, **kwargs: Any
+    ) -> Optional[schema.FeatureSettings]:
+        return schema.FeatureSettings.create(cls.name())
+
+    def reset_password(self, username: str, password: str) -> None:
+        # This uses the VMAccessForLinux extension to reset the credentials for an
+        # existing user or create a new user with sudo privileges.
+        # https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/vmaccess
+        reset_user_password = {"username": username, "password": password}
+        extension = self._node.features[AzureExtension]
+        result = extension.create_or_update(
+            name="VMAccessForLinux",
+            publisher="Microsoft.OSTCExtensions",
+            type_="VMAccessForLinux",
+            type_handler_version="1.5",
+            auto_upgrade_minor_version=True,
+            protected_settings=reset_user_password,
+        )
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+    def generate_password(self) -> str:
```

Comment:
I implemented a version, please pick it from my branch. It can be put in utils.

https://github.com/microsoft/lisa/commit/79569fd7f4c4f8e484ea9c79b72ca402e94a54eb

Reply:
Have picked. Added some changes because '\' and '-' are not allowed in the password according to the Azure doc.

Reply:
use cherry pick to pick the commit, instead of merging it into existing commits. You can reorder commits, to make sure the committed order is valid.